graphjet-core/src/test/java/com/twitter/graphjet/bipartite/MultiSegmentPowerLawBipartiteGraphTest.java

```diff
@@ -213,6 +213,21 @@ public void testMultiSegmentConstruction() throws Exception {
   }
 
   @Test
+  public void testMultiSegmentReverseIteration() throws Exception {
+    NodeMetadataLeftIndexedPowerLawMultiSegmentBipartiteGraph nodeMetadataLeftIndexedPowerLawMultiSegmentBipartiteGraph =
+      new NodeMetadataLeftIndexedPowerLawMultiSegmentBipartiteGraph(
+        3, 3, 5, 2, 2.0, 5, 2, new IdentityEdgeTypeMask(), new NullStatsReceiver());
+
+    addEdges(nodeMetadataLeftIndexedPowerLawMultiSegmentBipartiteGraph);
+
+    assertEquals(new LongArrayList(new long[]{13}),
```

Comment:
A more complicated test case is to create a randomly generated graph and compare the results of two iterators, which should be the exactly same. in the test case, it is better to check edge order across different segments as well as edge order within the same segment.